src/DataWrangling/inpaint_mask.jl

```diff
@@ -47,7 +47,7 @@ end
 propagate_horizontally!(field, ::Nothing, tmp_field=deepcopy(field); kw...) = field
 
 function propagating(field, mask, iter, max_iter)
-    mask_sum = sum(field; condition=mask)
+    mask_sum = sum(field; condition=interior(mask))
```

Comment:
?

Reply:
just checking a thing, apparently, we cannot use `condition::AbstractField` (because there is no `arch_array` method for fields). We could do two things: (1) extend the reductions for field conditions or (2) have an `arch_array` also for fields. I think the nice thing to do come up with a single function `on_architecture` that is valid for arrays, grids and fields alike

Reply:
agree that is needed. I guess there is an invalid assumption somewhere that `arch_array` works for `AbstractArray`.

Reply:
x-ref: https://github.com/CliMA/Oceananigans.jl/pull/3490